zip-codes/scripts/search.js

```diff
@@ -1,9 +1,45 @@
 'use strict';
 
 (function(module) {
-
+  webDB.execute(
+    'SELECT DISTINCT state FROM zips ' +
+    'ORDER BY state DESC;',
+    function(rows){
+      console.log(rows);
+      if (rows.length){
+        rows.forEach(function(a){
+          $('#state-select').append('<option>' + a.state + '</option>');
+        });
+      }
+      $('#state-select').on('change', function() {
+        $('#city-select').find('option').remove().end();
+        console.log($(this).val(), 'state value');
+        if ($(this).val()) {
+          webDB.execute(
+            'SELECT DISTINCT city FROM zips WHERE state="' + $(this).val() + '"' +
+            ' ORDER BY city DESC;',
+            function(rows){
+              if (rows.length){
+                rows.filter(function(dataEntry) {
+                  return $(this).val() === dataEntry.state;
+                })
+                .forEach(function(a){
+                  $('#city-select').append('<option>' + a.city + '</option>');
+                });
+              }
+            }
+          );
+        }
+      });
+    }
+  );
   // TODO: Write the code to populate your filters, and enable the search queries here in search.js
   // TODO: You will also interact with the map.js file here
+  // map = new google.maps.Map(document.getElementById("map"), {...});
```

Comment:
You'll have to target the form upon which a submit event was triggered, find its input box, and grab the value of that input box within the event handler. That'll give you your zip code of interest. Then, you can execute a SQL query with that zip code to find the latitude and longitude associated with that zip code. You could then call your initMap function and pass it that lat and lng (which can be used to make your marker).